src/gradient_aware_harmonisation/convergence.py

```diff
@@ -0,0 +1,335 @@
+            # we need gamma or 1-gamma, therefore we include the following condition
+            if self.apply_to_convergence:
+                return 1 - gamma
+            return gamma
+
+        if not isinstance(x, np.ndarray):
+            if x < self.initial_time:
+                if self.apply_to_convergence:
+                    return 0.0
+                return 1.0
+
+            if x >= self.final_time:
+                if self.apply_to_convergence:
+                    return 1.0
+                return 0.0
+
+            return decay(x)
+
+        # apply decay function only to values that lie between harmonisation
+        # time and convergence-time
+        x_gte_final_time = np.where(x >= self.final_time)
+        x_decay = np.logical_and(x >= self.initial_time, x < self.final_time)
+        gamma = np.ones_like(x, dtype=float)
+        gamma[x_gte_final_time] = 0.0
+        gamma[x_decay] = decay(x[x_decay])
+
+        if self.apply_to_convergence:
+            return 1 - gamma
+
+        return gamma
+
+    def derivative(self) -> CosineDecaySplineHelperDerivative:
+        """
+        Calculate the derivative of self
+
+        Returns
+        -------
+        :
+            Derivative of self
+        """
+        return CosineDecaySplineHelperDerivative(
+            initial_time=self.initial_time,
+            final_time=self.final_time,
+            apply_to_convergence=self.apply_to_convergence,
+        )
+
+    def antiderivative(self) -> SumOfSplines:
+        """
+        Calculate the anti-derivative/integral of self
+
+        Returns
+        -------
+        :
+            Anti-derivative of self
+        """
+        raise NotImplementedError
+
+
+def get_cosine_decay_harmonised_spline(
+    harmonisation_time: Union[int, float],
+    convergence_time: Union[int, float],
+    harmonised_spline_no_convergence: Spline,
+    convergence_spline: Spline,
+) -> SumOfSplines:
+    """
+    Generate the harmonised spline
+
+    Parameters
+    ----------
+    harmonisation_time
+        Harmonisation time
+
+        This is the time at and before which
+        the solution should be equal to `harmonised_spline_no_convergence`.
+
+    convergence_time
+        Convergence time
+
+        This is the time at and after which
+        the solution should be equal to `convergence_spline`.
+
+    harmonised_spline_no_convergence
+        Harmonised spline that does not consider convergence
+
+    convergence_spline
+        The spline to which the result should converge
+
+    Returns
+    -------
+    :
+        Harmonised spline
+    """
+    # The harmonised spline is considered as the spline that match
+    # the target-spline at the harmonisation time (wrt to zero-and
+    # first order derivative). Then we use a decay function to let
+    # the harmonised spline converge to the convergenve-spline (by
+    # default: harmonisee). This decay function has the form of a
```

Comment:
```suggestion
    # the harmonised spline converge to the convergence-spline. 
    # This decay function has the form of a
```

The harmonisee isn't defined within this context, so I would try and avoid referring to it